### DISCUSSION TITLE
BLE Power type

### DISCUSSION THREAD

**Discussion by mauroForlimpopoli** [Moddable-OpenSource/moddable#618] [Q&A]
Is there a command to read/write power of BLE and/or classic bluetooth or do I need to use native command?
At the moment I use this native command to read it:
esp_ble_tx_power_get(ESP_BLE_PWR_TYPE_DEFAULT);
and it works, but the command to set it :
esp_ble_tx_power_set(ESP_BLE_PWR_TYPE_DEFAULT, ESP_PWR_LVL_P9); 
doesn't work. I execute the instruction while the connection is on. What other I need to do? Thank you

**Comment by phoddie** [comment]
The BLE API does not currently provide a way to change the power, so you will need to use your own native function for that.

A quick search for `esp_ble_tx_power_set` shows that many developers don't observe a signal strength improvement. There doesn't seem to be a consistent reason for it.
